kotlinx-coroutines-core/common/src/NonCancellable.kt

```diff
@@ -29,6 +29,14 @@ public object NonCancellable : AbstractCoroutineContextElement(Job), Job {
 
     private const val message = "NonCancellable can be used only as an argument for 'withContext', direct usages of its API are prohibited"
 
+    /**
+     * Always returns `null`.
+     * @suppress **This an internal API and should not be used from general code.**
+     */
+    @Deprecated(level = DeprecationLevel.WARNING, message = message)
```

Comment:
Does it make sense to add `ReplaceWith(null)` to allow for automatic replacements in bulk? I don't see the downsides.

Reply:
Not really, it rather means that the whole usage of this API is incorrect and it's time to rethink what is written :)